src/wp-includes/class-wp-theme-json.php

```diff
@@ -1000,6 +1000,14 @@ protected static function get_blocks_metadata() {
 		$registry = WP_Block_Type_Registry::get_instance();
 		$blocks   = $registry->get_all_registered();
 
+		// Unset old blocks from static variable.
+		$blocks_to_unset = array_diff_key( static::$blocks_metadata, $blocks );
+		if ( ! empty( $blocks_to_unset ) ) {
+			foreach ( $blocks_to_unset as $block_name => $block_meta ) {
+				unset( static::$blocks_metadata[ $block_name ] );
+			}
+		}
+
```

Comment:
We haven't been clearing unregistered blocks till now. This caused inconsistent outcomes, especially when the cache was in place.